Заменить сохранение локальных переменных на обработку потока, в шаблоне использовать asyncPipe

sinelamin opened the pull request.

sinelamin committed:
feat(core): install Angular Material and prepare application structure

Install Angular Material and set up the basic application structure

sinelamin committed:
feat(core): add basic layout for app

add components tasks-list, task-item, todo-input, main, basic ui structure for components, interface for task and service with mock data

sinelamin committed:
feat(task): create basic layout for task-item

Create the basic layout for the task-item component.
Add styles and state handling for task completion.
Introduce a model for task representation.

sinelamin committed:
Merge pull request #1 from sinelamin/basic-ui

Basic UI

sinelamin committed:
feat(tasks): add methods for task management

Implement methods for managing tasks:
- getTasks(): returns the list of tasks as an observable.
- addTask(task): adds a new task to the list and updates the state.
- deleteTask(task): removes a task from the list and updates the state.

sinelamin committed:
feat(tasks): add task filtering, status management, and task counter

Add toggle buttons for filtering the task list.
Implement filterTasks() to update the task list based on filter selection.
Add changeStatusTask() to update task status when checkbox is checked.
Add tasksCounter() method to track the number of incomplete tasks.
Display incomplete task count in template of tasks-list.
Refactor task list rendering to use filteredTasks instead of tasks.
Minor adjustments to existing code to support new functionality.

sinelamin committed:
Merge pull request #2 from sinelamin/task-management

Task management

sinelamin committed:
feat(form): add validation for empty field

add validation to prevent submission of empty fiel in the form

sinelamin committed:
refactor(material): Transfer all imported modules angular material to the re-export module

Created a new module to re-export all angular material modules
Replaced all imported modules angular material with the new re-export module

Commit:
refactor(tasks): remove TaskModel, rename TaskInterface, change delete logic

Removed TaskModel
Renamed TaskInterface to Task
Changed property status to completed and add property id
The logic for deleting tasks has been changed: deleting by id, and not by reference to an object

Commit:
refactor(tasks): change formGroup variable name to inputChangeTaskName

tamazyanarsen commented on src/app/feature/tasks/components/task-item/task-item.component.ts:
```diff
@@ -0,0 +1,60 @@
+})
+export class TaskItemComponent implements OnInit {
+
+  @Input() task!: Task;
```
убрать `!`
not null assertion

sinelamin replied:
исправил

Commit:
refactor(task): убрал оператор ! для значений

Убрал оператор ! и добавил проверку на null и undefined для значений в task-item

Commit:
refactor(task): tasksCounter

Теперь возвращает строку с кол-вом не завершенных задач, в единственном числе для 0 и 1 задач

Commit:
feat(tasks): кнопка и метод для изменения статуса всех задач

Добавил кнопку в шаблон компонента todo-input и метод в task.service для изменения статуса всех задач

Commit:
feat(task): кнопка удаления всех выполненных задач

метод deleteAllComplitedTasks для удаления всех выполенных задач
поток hasCompletedTasks и метод checkingUnfinishedTasks для проверки наличия выполненных задач

Commit:
fix(task): метод changeStatusAllTasks

Теперь метод изменяет все задачи либо на true либо на false

tamazyanarsen commented on src/app/feature/todo-input/todo-input.component.ts:
```diff
@@ -0,0 +1,48 @@
+
+  ngOnInit(): void {
+    this.taskService.tasks$.subscribe((tasks => {
+      this.tasks = tasks;
```
не нужно отдельно сохранять в свойство класса данные из сервиса

оставляй в виде потока, а в html используй asyncPipe

sinelamin replied:
Исправил

Commit:
refactor(todo-input): заменить сохранение данных из сервиса на поток с использованием asyncPipe

Убрано сохранение данных из сервиса в свойство класса
taskService сделал публичным
Данные обрабатываются напрямую из потока
В шаблоне использован asyncPipe

Commit:
refactor(tasks-list): заменить сохранение данных из сервиса на обработку потока

Данные обрабатываются напрямую из потока
В шаблоне использован asyncPipe
